libraries/common/io.effekt

```diff
@@ -6,49 +6,80 @@ import queue
 extern include llvm "../llvm/io.ll"
 extern include js "../js/io.js"
 
-// Asynchronous I/O
-// ----------------
-
-interface AsyncIO {
-  /**
-   * Runs the provided computation and conceptually waits until it completes by
-   * calling the writer-continuation.
-   *
-   * The provided writer-continuation should only ever be called ONCE.
-   * The computation is passed as a thunk in order to prevent accidental usage of
-   * other control effects that would violate the contract of calling the writer
-   * once.
-   */
-  def performIO[T](computation: (T => Unit at {io, global}) => Unit at {io, global}): T
-}
 
-/**
-  * Run an IO computation (that uses callbacks) here.
-  */
-def blocking(program: () => Unit / AsyncIO at {io, global}): Unit =
-  try { program() }
-  with AsyncIO {
-    def performIO[T](computation) = computation(resume)
-  }
+// Event Loop
+// ----------
+
+interface Eventloop {}
+
+type Task[T] = () => T / Eventloop at {io, control, global}
+
+def eventloop(program: Task[Unit]): Unit = direct::eventloop(box {
+  try program() with Eventloop {}
+})
```

Comment:
As discussed in the meeting, let's drop this effect and automatically insert `eventloop` when `main` has capture `control`.